### ISSUE TITLE
Staging travis-CI failing, Jenkins unstable, post 3.4.7

### ISSUE THREAD

**Issue by photodude** [joomla/joomla-cms#8778]
Something in the 3.4.7 changes started travis-CI to fail and put Jenkins status to unstable
http://build.joomla.org/job/cms/2913/
https://travis-ci.org/joomla/joomla-cms/jobs/98573432

`PHP Fatal error:  Call to a member function get() on null in /libraries/joomla/session/session.php on line 500`
```
PHP Stack trace:
PHP   1. {main}() /libraries/vendor/phpunit/phpunit/phpunit:0
PHP   2. PHPUnit_TextUI_Command::main() /libraries/vendor/phpunit/phpunit/phpunit:47
PHP   3. PHPUnit_TextUI_Command->run() /libraries/vendor/phpunit/phpunit/src/TextUI/Command.php:100
PHP   4. PHPUnit_TextUI_TestRunner->doRun() /libraries/vendor/phpunit/phpunit/src/TextUI/Command.php:148
PHP   5. PHPUnit_Framework_TestSuite->run() /libraries/vendor/phpunit/phpunit/src/TextUI/TestRunner.php:436
PHP   6. PHPUnit_Framework_TestSuite->run() /libraries/vendor/phpunit/phpunit/src/Framework/TestSuite.php:735
PHP   7. PHPUnit_Framework_TestSuite->run() /libraries/vendor/phpunit/phpunit/src/Framework/TestSuite.php:735
PHP   8. PHPUnit_Framework_TestCase->run() /libraries/vendor/phpunit/phpunit/src/Framework/TestSuite.php:735
PHP   9. PHPUnit_Framework_TestResult->run() /libraries/vendor/phpunit/phpunit/src/Framework/TestCase.php:705
PHP  10. PHPUnit_Framework_TestCase->runBare() /libraries/vendor/phpunit/phpunit/src/Framework/TestResult.php:601
PHP  11. PHPUnit_Framework_TestCase->runTest() /libraries/vendor/phpunit/phpunit/src/Framework/TestCase.php:749
PHP  12. ReflectionMethod->invokeArgs() /libraries/vendor/phpunit/phpunit/src/Framework/TestCase.php:889
PHP  13. JComponentRouterViewTest->testGetPath() /libraries/vendor/phpunit/phpunit/src/Framework/TestCase.php:889
PHP  14. JComponentRouterView->getPath() /tests/unit/suites/libraries/cms/component/router/JComponentRouterViewTest.php:139
PHP  15. call_user_func_array:{/libraries/cms/component/router/view.php:116}() /libraries/cms/component/router/view.php:116
PHP  16. JComponentRouterViewInspector->getCategorySegment() /libraries/cms/component/router/view.php:116
PHP  17. JCategories->get() /libraries/cms/component/router/stubs/JComponentRouterViewInspector.php:59
PHP  18. JCategories->_load() /libraries/legacy/categories/categories.php:184
PHP  19. JFactory::getUser() /libraries/legacy/categories/categories.php:214
PHP  20. JSession->get() /libraries/joomla/factory.php:236
```

**Comment by creativeprogramming** [comment]
See here:

https://github.com/joomla/joomla-cms/compare/3.4.6...3.4.7#L430

in 3.4.6 it was:

```php
        public function get($name, $default = null, $namespace = 'default')
 	{
 	        //...omissis...
		if (isset($_SESSION[$namespace][$name]))
		{
			return $_SESSION[$namespace][$name];
		}
		return $default;
      }
```

in 3.4.7 is:

```php
 	public function get($name, $default = null, $namespace = 'default')
 	{
 		//...omissis...

		return $this->data->get($namespace . '.' . $name, $default);
 	}
```

so, we are missing the isset() now, and as $this->data is not initialized, so we should do something like:

```php
 	public function get($name, $default = null, $namespace = 'default')
 	{
 		//...omissis...
               if ($this->data!=null)
               {
                   return $this->data->get($namespace . '.' . $name, $default);
               }
	       return $default;
 	}
```

or initialize the $data property to Registry

```php
protected $data = new \Joomla\Registry\Registry;
```

and then rely on the Registry->get() for the default value

**Comment by Fedik** [comment]
there was fix already #8773 no?

**Comment by nikosdion** [comment]
Guys, it's already fixed https://github.com/joomla/joomla-cms/pull/8773 A new version 3.4.8 will be released as soon as possible, as far as I'm told (I'm not in any official position, just helping in the 3.4.7 cleanup).

**Comment by photodude** [comment]
@wilsonge I disagree with closing this. This issue is specific to Travis-ci failing, until there is a PR merged in the staging branch that resolves that, IMO this should remain open. (secondary is the jenkins stability issue)

Consider that every PR based on the current staging branch will also fail in Travis-CI, this greatly slows down reviewing PR's in GitHub. It also makes people even more complacent to their PR's failing, and could even prevent people from noticing the issues in their PR's.

**Comment by wilsonge** [comment]
I've merged in 3.4.8 - travis now passes :)

**Comment by photodude** [comment]
Thanks @wilsonge Much better having a stable Travis-CI and Jenkins in the staging branch.
